source/docs/software/commandbased/builtins.rst

```diff
@@ -0,0 +1,379 @@
+
+.. tabs::
+
+  .. code-tab:: java
+
+    new PrintCommand("This message will be printed!")
+
+  .. code-tab:: c++
+
+    frc2::PrintCommand("This message will be printed!")
+
+
+Dynamically Deciding What Command To Schedule
+---------------------------------------------
+
+Sometimes it's desired to run a command out of a few options based on sensor feedback or other data known only at runtime.
+
+SelectCommand
+^^^^^^^^^^^^^
+
+For this, ``SelectCommand`` (`Java <https://first.wpi.edu/wpilib/allwpilib/docs/release/java/edu/wpi/first/wpilibj2/command/SelectCommand.html>`__, `C++ <https://first.wpi.edu/wpilib/allwpilib/docs/release/cpp/classfrc2_1_1_select_command.html>`__) accepts a map of commands and a generic selector, or a ``Supplier<Command>``.
+
+.. note:: While the Java version of ``SelectCommand`` simply uses an ``Object`` as a key, the C++ version is templated on the key type.
+
+.. note:: An alternate version of ``SelectCommand`` simply takes a method that supplies the command to be run - this can be very succinct, but makes inferring the command’s requirements impossible, and so leaves the user responsible for manually adding the requirements to the SelectCommand.
+
+.. tabs::
+
+  .. group-tab:: Java
+
+    .. remoteliteralinclude:: https://raw.githubusercontent.com/wpilibsuite/allwpilib/v2022.4.1/wpilibjExamples/src/main/java/edu/wpi/first/wpilibj/examples/selectcommand/RobotContainer.java
+      :language: java
+      :lines: 20-45
+      :linenos:
+      :lineno-start: 20
+
+  .. group-tab:: C++ (Header)
+
+    .. remoteliteralinclude:: https://raw.githubusercontent.com/wpilibsuite/allwpilib/v2022.4.1/wpilibcExamples/src/main/cpp/examples/SelectCommand/include/RobotContainer.h
+      :language: c++
+      :lines: 25-44
+      :linenos:
+      :lineno-start: 25
+
+ConditionalCommand
+^^^^^^^^^^^^^^^^^^
+
+``ConditionalCommand`` (`Java <https://first.wpi.edu/wpilib/allwpilib/docs/release/java/edu/wpi/first/wpilibj2/command/ConditionalCommand.html>`__, `C++ <https://first.wpi.edu/wpilib/allwpilib/docs/release/cpp/classfrc2_1_1_conditional_command.html>`__) is a specialized version of this that decides between two commands using a boolean condition.
+
+.. tabs::
+
+  .. code-tab:: java
+
+    // Runs either commandOnTrue or commandOnFalse depending on the value of m_limitSwitch.get()
+    new ConditionalCommand(commandOnTrue, commandOnFalse, m_limitSwitch::get)
+
+  .. code-tab:: c++
+
+    // Runs either commandOnTrue or commandOnFalse depending on the value of m_limitSwitch.get()
+    frc2::ConditionalCommand(commandOnTrue, commandOnFalse, [&m_limitSwitch] { return m_limitSwitch.Get(); })
+
+
+Waiting For Delays
+------------------
+
+Wait for a certain condition to happen or adding a delay can be useful to synchronize between different commands in a command group or between other robot actions.
```

Comment:
I'm confused. Is `Wait` supposed to be a decorator? If so, wrap it in an inline code block such as ``Wait()``. I presume it means `WaitCommand()`. Either way, this sentence is confusing.